commands/sampledata.go

```diff
@@ -0,0 +1,840 @@
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/utils"
+
+	"github.com/icrowley/fake"
+	"github.com/spf13/cobra"
+)
+
+const (
+	deactivatedUser = "deactivated"
+	guestUser       = "guest"
+	attachmentsDir  = "attachments"
+)
+
+var SampledataCmd = &cobra.Command{
+	Use:   "sampledata",
+	Short: "Generate sample data",
+	Long:  "Generate a sample data file and store it locally, or directly import it to the remote server",
+	Example: `  # you can create a sampledata file and store it locally
+  $ mmctl sampledata --bulk sampledata-file.jsonl
+
+  # or you can simply print it to the stdout
+  $ mmctl sampledata --bulk -
+
+  # the amount of entities to create can be customized
+  $ mmctl sampledata -t 7 -u 20 -g 4
+
+  # the sampledata file can be directly imported in the remote server by not specifying a --bulk flag
+  $ mmctl sampledata
+
+  # and the sample users can be created with profile pictures
+  $ mmctl sampledata --profile-images`,
+	Args: cobra.NoArgs,
+	RunE: withClient(sampledataCmdF),
+}
+
+func init() {
+	SampledataCmd.Flags().Int64P("seed", "s", 1, "Seed used for generating the random data (Different seeds generate different data).")
+	SampledataCmd.Flags().IntP("teams", "t", 2, "The number of sample teams.")
+	SampledataCmd.Flags().Int("channels-per-team", 10, "The number of sample channels per team.")
+	SampledataCmd.Flags().IntP("users", "u", 15, "The number of sample users.")
+	SampledataCmd.Flags().IntP("guests", "g", 1, "The number of sample guests.")
+	SampledataCmd.Flags().Int("deactivated-users", 0, "The number of deactivated users.")
+	SampledataCmd.Flags().Int("team-memberships", 2, "The number of sample team memberships per user.")
+	SampledataCmd.Flags().Int("channel-memberships", 5, "The number of sample channel memberships per user in a team.")
+	SampledataCmd.Flags().Int("posts-per-channel", 100, "The number of sample post per channel.")
+	SampledataCmd.Flags().Int("direct-channels", 30, "The number of sample direct message channels.")
+	SampledataCmd.Flags().Int("posts-per-direct-channel", 15, "The number of sample posts per direct message channel.")
+	SampledataCmd.Flags().Int("group-channels", 15, "The number of sample group message channels.")
+	SampledataCmd.Flags().Int("posts-per-group-channel", 30, "The number of sample posts per group message channel.")
+	SampledataCmd.Flags().String("profile-images", "", "Optional. Path to folder with images to randomly pick as user profile image.")
+	SampledataCmd.Flags().StringP("bulk", "b", "", "Optional. Path to write a JSONL bulk file instead of uploading into the remote server.")
+
+	RootCmd.AddCommand(SampledataCmd)
+}
+
+func randomPastTime(seconds int) int64 {
```

Comment:
nit: In order to keep consistency across the code base, we can move utility functions to `commands/sampledata_util.go`.